src/quark-core/src/QuarkScript.sol

```diff
@@ -17,14 +17,24 @@ abstract contract QuarkScript {
 
     /// @notice A safer, but gassier reentrancy guard that writes the flag to the QuarkStateManager
     modifier nonReentrant() {
-        if (read(REENTRANCY_FLAG_SLOT) == bytes32(uint256(1))) {
+        bytes32 slot = REENTRANCY_FLAG_SLOT;
+        bytes32 flag;
+        // TODO: Move to TSTORE after updating Solidity version to >=0.8.24
+        assembly {
+            flag := sload(slot)
+        }
```

Comment:
It's not using `read(...)` here because this will use TLOAD while `read(...)` will be an SLOAD.

I don't think there's much use-cases for scripts to use `tstore/tload` outside of re-entrancy guards. `sstore/sload` will be more commonly used, since most of the state that scripts read/write will need to be persisted across transactions.

Reply:
If storage must be used by a script, I think we should still expose a "safe" way to do it. Scripts writing to declared storage variables will almost certainly result in storage collisions. The cool thing about our new nonce model is that we could have some guarantee of isolated storage *if* you always use the storage helpers. I left a TODO for it, but the `read/write` helpers could `TLOAD` the nonce and use that to create nonce-isolated keys.